samza-core/src/main/java/org/apache/samza/execution/StreamEdge.java

```diff
@@ -128,12 +121,13 @@ Config generateConfig() {
       config.put(String.format(StreamConfig.CONSUMER_OFFSET_DEFAULT_FOR_STREAM_ID(), spec.getId()), "oldest");
       config.put(String.format(StreamConfig.PRIORITY_FOR_STREAM_ID(), spec.getId()), String.valueOf(Integer.MAX_VALUE));
     }
-    if (spec.isBounded()) {
-      config.put(String.format(StreamConfig.IS_BOUNDED_FOR_STREAM_ID(), spec.getId()), "true");
```

Comment:
Did this get moved somewhere else?

Reply:
No, I removed it. AFAICT, this is only set by AbstractApplicationRunner when creating a StreamSpec, and it does so by reading it from the config:
AbstractApplicationRunner#getStreamSpec(String, String, String):
```
boolean isBounded = streamConfig.getIsBounded(streamId);
```
StreamEdge is just reading it back from the spec and writing to the config for the stream.

@xinyuiscool can you confirm this is safe?

Reply:
So, I think that it is a bit confusing here to me, regarding to where we specify a stream to be bounded as a single place for the source of truth: is it in the user config for a stream? is it in StreamSpec? And what's the relationship between user defined stream configuration and StreamSpec? I thought that the relationship is StreamSpec will be translating to stream configuration while submitting the job. So, it seems that config is the *derived* data set from *StreamSpec*. Hence, all stream-specific properties should be found in the SoT object (i.e. StreamSpec). Could you elaborate what do you consider as the SoT object for a stream in the coming change? StreamDescriptor?StreamSpec?Or StreamConfig? And what are the relationship between them? Thanks!

Reply:
it would be more clear if we know in step 2 and 4, which config takes precedence in the merge. And since it is a complex logic here, it would be nice to document it somewhere. Maybe in ExecutionPlanner or JobNode.generateConfig()?